bears/vcs/git/GitCommitBear.py

```diff
@@ -203,3 +223,80 @@ def check_body(self, body,
             yield Result(self, 'Body of HEAD commit contains too long lines. '
                                'Commit body lines should not exceed {} '
                                'characters.'.format(body_line_length))
+
+    def check_issue(self, body,
+                    body_regex: str=None,
+                    body_close_issue: bool=False,
+                    body_close_issue_url: bool=False,
+                    body_close_issue_last_line: bool=False):
+        """
+        Check for matching issue related URLs.
+
+        :param body:
+            Body of the commit message of HEAD.
+        :param body_regex:
+            If provided, matches the keywords with the regex. This takes
+            precedence over body_close_issue.
+        :param body_close_issue:
+            Whether to automatically detect host from git config or not.
+            If detected checks for the presence of issue close voodoo
+            within the commit body. By default, if none of
+            body_close_issue_url and body_close_issue_last_line are enabled,
+            checks for presence of short references like 'closes #213', etc.
+        :param body_close_issue_url:
+            Checks the presence of issue close reference with a full URL
+            related to some issue. Works along with body_close_issue.
+        :param body_close_issue_last_line:
+            When enabled, checks for issue close reference presence on the
+            last line of the commit body. Works along with body_close_issue.
+        """
+        if body_regex:
+            if not re.search(body_regex, body):
+                yield Result(self, 'No match found for the regular '
+                                   'expression provided: %s' % body_regex)
+            return
+        if not body_close_issue:
+            return
+        remotes, _ = run_shell_command('git config --get-regex url')
+        remotes = [url.split()[-1] for url in remotes.splitlines()]
+        host = None
+
+        for url in remotes:
+            netloc = urlparse(url)[2]
+            if 'git@' in url:
+                netloc = re.findall(r'@(\S+):', url)[0]
+            else:
+                netloc = urlparse(url)[1]
+            host = netloc.split('.')[0]
+        if host not in self.SUPPORTED_HOSTS:
+            return
+        keys = self.SUPPORTED_HOSTS[host]
+        body = body.lower()
+        good_ref = False
+        result = 'No {} reference found in the body at HEAD commit.'
+
+        if body_close_issue_last_line:
+            body = body.splitlines()[-1]
+            result = ('No {} reference found in the body at HEAD commit in '
+                      'the last line.')
```

Comment:
rather use an else clause instead of this override pattern:
```
if body_close_issue_last_line:
    body = ...
    result = ...
else:
    result = 'No {} reference found in the body at HEAD commit.'
```